Android/app/src/main/java/utils/GPSTracker.java

```diff
@@ -56,7 +59,10 @@ private Location getLocation() {
 
             if (!isGPSEnabled && !isNetworkEnabled) {
                 // no network provider is enabled
-            } else {
+            } else if (ActivityCompat.checkSelfPermission(this, Manifest.permission.ACCESS_FINE_LOCATION)
+                    == PackageManager.PERMISSION_GRANTED
+                    && ActivityCompat.checkSelfPermission(this, Manifest.permission.ACCESS_COARSE_LOCATION)
+                    == PackageManager.PERMISSION_GRANTED) {
```

Comment:
Shouldnt we ask for permissions if they are not granted?

Reply:
We are asking for permissions where this service is called, without giving the permission user can not proceed. In this service we are just checking if the permission is given(we can not call locationManager.requestLocationUpdates without checking if permissions are given or not).